packages/terminal/src/browser/terminal-info-toolbar-item.tsx

```diff
@@ -0,0 +1,113 @@
+// *****************************************************************************
+// Copyright (C) 2023 STMicroelectronics and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+import { ReactTabBarToolbarItem } from '@theia/core/lib/browser/shell/tab-bar-toolbar';
+import { ReactNode } from '@theia/core/shared/react';
+import React = require('@theia/core/shared/react');
+import { HoverService, Widget } from '@theia/core/lib/browser';
+import { TerminalWidget } from './base/terminal-widget';
+import { MarkdownRenderer } from '@theia/core/lib/browser/markdown-rendering/markdown-renderer';
+import { MarkdownStringImpl } from '@theia/core/lib/common/markdown-rendering/markdown-string';
+import { DisposableCollection } from '@theia/core';
+
+export class TerminalInfoToolbarItem implements ReactTabBarToolbarItem {
+    readonly id = 'terminal:info';
+
+    constructor(
+        protected readonly hoverService: HoverService,
+        protected readonly markdownRenderer: MarkdownRenderer
+    ) {}
+
+    isVisible(widget?: Widget): boolean {
+        return widget instanceof TerminalWidget;
+    }
+
+    render(widget?: Widget): ReactNode {
+        const toDispose = new DisposableCollection();
+        return (
+            <div
+                id={this.id}
+                className='codicon codicon-terminal-bash action-label'
+                onMouseEnter={e => this.onMouseEnter(e, toDispose, widget)}
+                onMouseLeave={e => this.onMouseLeave(toDispose)}
+            ></div>
+        );
+    }
+
+    protected async onMouseEnter(
```

Comment:
It seems odd that we have to implement our own "show hover on mouse-over" support. Dont' we have this somewhere else already?

Reply:
Not that I am aware of. I think the reusable functionality is all in the `HoverService`. However we have to call `hoverService.requestHover` from somewhere. Here we have drawn the `div` over which the hover will appear ourself and the `onMouseEnter` only creates the custom hover ui and then calls the Hover Service. I don't think we could omit much of this.

Reply:
Code removed because of usage of enhanced preview